apps/meteor/client/views/admin/users/UsersTable/UsersTableRow.tsx

```diff
@@ -42,6 +54,71 @@ const UsersTableRow = ({ user, onClick, mediaQuery, tab }: UsersTableRowProps):
 		.filter((roleName): roleName is string => !!roleName)
 		.join(', ');
 
+	const userId = user._id;
+	const isAdmin = user.roles?.includes('admin');
+	const isActive = user.active;
+	const isFederatedUser = isUserFederated(user);
+
+	const onChange = useEffectEvent(() => {
+		onReload();
+	});
+
+	const changeAdminStatusAction = useChangeAdminStatusAction(userId, isAdmin, onChange);
+	const changeUserStatusAction = useChangeUserStatusAction(userId, isActive, onChange);
+	const deleteUserAction = useDeleteUserAction(userId, onChange, onReload);
+	const resetTOTPAction = useResetTOTPAction(userId);
+	const resetE2EKeyAction = useResetE2EEKeyAction(userId);
+	const resendWelcomeEmail = useSendWelcomeEmailMutation();
+
+	const isNotPendingDeactivatedNorFederated = tab !== 'pending' && tab !== 'deactivated' && !isFederatedUser;
+	const menuOptions = {
+		...(isNotPendingDeactivatedNorFederated &&
+			changeAdminStatusAction && {
+				makeAdmin: {
+					label: { label: changeAdminStatusAction.label, icon: changeAdminStatusAction.icon },
+					action: changeAdminStatusAction.action,
+				},
+			}),
+		...(isNotPendingDeactivatedNorFederated &&
+			resetE2EKeyAction && {
+				resetE2EKey: { label: { label: resetE2EKeyAction.label, icon: resetE2EKeyAction.icon }, action: resetE2EKeyAction.action },
+			}),
+		...(isNotPendingDeactivatedNorFederated &&
+			resetTOTPAction && {
+				resetTOTP: { label: { label: resetTOTPAction.label, icon: resetTOTPAction.icon }, action: resetTOTPAction.action },
+			}),
+		...(changeUserStatusAction &&
+			!isFederatedUser && {
+				changeActiveStatus: {
+					label: { label: changeUserStatusAction.label, icon: changeUserStatusAction.icon },
+					action: changeUserStatusAction.action,
+				},
+			}),
+		...(deleteUserAction && {
+			delete: { label: { label: deleteUserAction.label, icon: deleteUserAction.icon }, action: deleteUserAction.action },
+		}),
+	};
+
+	const handleResendWelcomeEmail = () => {
+		if (!emails?.length) {
+			dispatchToastMessage({ type: 'error', message: t('Welcome_email_failed') });
+			return;
+		}
```

Comment:
Thist verification is avoiding undefined for the emails list because the emails field of IUser is optional, if it is not possible for it not to be set why is the field optional?